### DISCUSSION TITLE
Where does OTBR store it's state?

### DISCUSSION THREAD

**Discussion by olavt** [openthread/openthread#9699] [Q&A] [answered]
Where does the Open Thread Border Router store it's state?

What if I need to move the state over to another computer, or if I need to backup / restore the state?

**Comment by jwhui** [comment]
The settings path is defined by `OPENTHREAD_CONFIG_POSIX_SETTINGS_PATH`.

https://github.com/openthread/openthread/blob/ae6eff5f212d88d86448d4c085d52098babe13a9/src/posix/platform/settings.cpp#L92

**Reply by olavt** [reply]
What happens when you swap a Radio Co-Processor on a OpenThread Border Router? Do you have to form a new network after replacing the RCP, or should the new RCP just work with the existing network?

**Reply by jwhui** [reply]
Given that state is stored on the host, you can swap the RCP without needing to factory reset. Note that the path is based on IEEE EUI-64, so if that changes, then the settings file also will be different.

**Comment by olavt** [comment]
How can I see the current settings path of the OpenThread Border Router when I'm logged in to the host OS?

**Reply by jwhui** [reply]
As mentioned in https://github.com/orgs/openthread/discussions/9699#discussioncomment-7805050, it is a build-time configuration with `OPENTHREAD_CONFIG_POSIX_SETTINGS_PATH`.

**Reply by olavt** [reply]
So, it’s not possible to know where the settings location is on a running system? Is there a settings backup / restore command available? Any documented process for backup / restore of thebOTBR state?

**Reply by jwhui** [reply]
On Linux, you can list the set of open file descriptors:

https://unix.stackexchange.com/questions/333186/how-to-list-the-open-file-descriptors-and-the-files-they-refer-to-in-my-curren

**Reply by olavt** [reply]
What is the default location if I build OTBR on a Raspberry Pi with Raspberry Pi OS?

**Reply by jwhui** [reply, marked as answer]
https://github.com/openthread/ot-br-posix/blob/ab38f784f647e23952033b7f44b9ed7779835a08/third_party/openthread/CMakeLists.txt#L96
